lib/src/public_client_application.dart

```diff
@@ -88,7 +90,17 @@ class PublicClientApplication {
         return MsalInitializationException();
       case "AUTH_ERROR":
       default:
-        return MsalException("Authentication error");
+
+        /// PlatformException(AUTH_ERROR, Authentication error, The operation couldn’t be completed. (MSALErrorDomain error -50000.))
+        if (e.details != null &&
+            e.details
+                .toString()
+                .toLowerCase()
+                .contains('msalerrordomain error -50000')) {
```

Comment:
Thanks for the pr, but as far as i'm aware msalerrordomain -50000 is not a forgot password error but a general error in MSAL and can come up in several different scenarios.

Reply:
For some reason, IOS returns -50000 when forgot password is tapped on Microsoft login form.